### DISCUSSION TITLE
Writing integration tests for Next.js pages

### DISCUSSION THREAD

**Discussion by josch87** [neuefische/questions#648] [Web] [answered]
I renamed a field in my data structure (from "date" to "dateOfInteraction". Therefore a sort on the [InteractionsPage](https://github.com/josch87/Gunther/blob/10e9118713a7cd6a5d5a44b447966f27a5ee5530/pages/interactions/index.js#L5-L14) did not work anymore without throwing an error. I encountered this only by testing the frontend.

How can I write integration tests on a page in order to find out about such errors? Or are there other ways to get attention of such errors?

**Comment by actyvystom** [comment, marked as answer]
Hi @josch87,
you could set up `unit tests` for that purpose (and combine several unit tests to an integration test). But it would imply that you outsource  your functions to be tested so that you can set up corresponding testing files. 
Please check the [unit testing handout](https://github.com/neuefische/bo-web-23-3/blob/main/sessions/js-unit-testing/js-unit-testing.md) of your class repository for further details on setting up unit tests. 
💡 As we only focus on component testing during capstone, this is **not** a requirement for capstone projects (although an honorable goal! 🎖️).
Cheers, Thomas
